experimental/packages/sdk-logs/src/LoggerProvider.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { diag } from '@opentelemetry/api';
+import type * as logsAPI from '@opentelemetry/api-logs';
+import { IResource, Resource } from '@opentelemetry/resources';
+import { getEnv, merge } from '@opentelemetry/core';
+
+import type { LoggerProviderConfig, LogRecordLimits } from './types';
+import type { LogRecordProcessor } from './LogRecordProcessor';
+import type { LogRecordExporter } from './export/LogRecordExporter';
+import { Logger } from './Logger';
+import { loadDefaultConfig } from './config';
+import { MultiLogRecordProcessor } from './MultiLogRecordProcessor';
+import { BatchLogRecordProcessor } from './platform/node/export/BatchLogRecordProcessor';
+import { NoopLogRecordProcessor } from './export/NoopLogRecordProcessor';
+
+export type EXPORTER_FACTORY = () => LogRecordExporter;
+
+export class LoggerProvider implements logsAPI.LoggerProvider {
+  protected static readonly _registeredExporters = new Map<
+    string,
+    EXPORTER_FACTORY
+  >();
+
+  public readonly resource: IResource;
+
+  private readonly _loggers: Map<string, Logger> = new Map();
+  private readonly _logRecordLimits: LogRecordLimits;
+  private _activeProcessor: MultiLogRecordProcessor;
+  private readonly _registeredLogRecordProcessors: LogRecordProcessor[] = [];
+  private readonly _forceFlushTimeoutMillis;
+
+  constructor(config: LoggerProviderConfig = {}) {
+    const { resource, logRecordLimits, forceFlushTimeoutMillis } = merge(
+      {},
+      loadDefaultConfig(),
+      config
+    );
+    this.resource = Resource.default().merge(resource ?? Resource.empty());
+    this._logRecordLimits = logRecordLimits;
+    this._forceFlushTimeoutMillis = forceFlushTimeoutMillis;
+
+    const defaultExporter = this._buildExporterFromEnv();
+    if (defaultExporter !== undefined) {
+      this._activeProcessor = new MultiLogRecordProcessor(
+        [new BatchLogRecordProcessor(defaultExporter)],
+        forceFlushTimeoutMillis
+      );
+    } else {
+      this._activeProcessor = new MultiLogRecordProcessor(
```

Comment:
Do we need to use the multi processor here Could we just assign the Noop processor instead? Same for the Batch processor above.

Reply:
The main purpose is to use the `MultiLogRecordProcessor` to manage `forceFlush` and `shutdown`, especially `forceFlush` timeout handling.

There are some differences with the `TraceProvider`: 
In `TraceProvider`, `forceFlush` is handled through `_registeredSpanProcessors`， and the defaultExporter handling, (use `BatchLogRecordProcessor` & `NoopLogRecordProcessor`), is not added to `_registeredSpanProcessors`. This will cause `forceFlush` not work ,  I'm not sure if this is a bug or specification？  can discuss it together

Reply:
I don't think the timeout handling matters with NoopLogRecordProcessors since it resolves immediately. But I don't think this needs to hold the review.

Reply:
> I don't think the timeout handling matters with NoopLogRecordProcessors since it resolves immediately. But I don't think this needs to hold the review.

Yeah, the NoopLogRecordProcessor doesn't have to do timeout processing.  But here it is expected to be handled by the `MultiLogRecordProcessor`, otherwise, it requires additional judgment as `NoopLogRecordProcessor`. I think the way it's done now is probably a little easier to understand .